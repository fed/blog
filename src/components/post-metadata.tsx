import React from 'react';
import styled from 'styled-components';

import calendarIcon from '../assets/calendar.svg';
import categoryIcon from '../assets/category.svg';
import clockIcon from '../assets/clock.svg';
import { sizeContainerMedium, fontFamilySansSerif, sizeContainerLarge } from '../styles/mixins';

const List = styled.ul`
    background-color: #deebff;
    border-radius: 3px;
    list-style-type: none;
    margin: 0;
    padding: 16px;
    border-left: 6px solid #4c9aff;
    @media (min-width: ${sizeContainerLarge}) {
        align-items: center;
        display: flex;
    }
`;

const ListItem = styled.li`
    font-family: ${fontFamilySansSerif};
    font-size: 16px;
    font-weight: 200;
    display: flex;
    :not(:last-child) {
        margin-right: 24px;
        margin-bottom: 16px;
    }
    @media (min-width: ${sizeContainerMedium}) {
        align-items: center;
        display: inline-flex;
        margin-bottom: 0;
        :not(:last-child) {
            margin-right: 40px;
            margin-bottom: 0;
        }
    }
`;

const Icon = styled.img`
    height: 20px;
    margin-right: 8px;
    width: 20px;
    @media (min-width: ${sizeContainerLarge}) {
        margin-right: 10px;
    }
`;

interface Props {
    date: string;
    categoryTitle: string;
    timeToRead: number;
}

export const PostMetadata: React.FC<Props> = ({ date, categoryTitle, timeToRead }) => (
    <List>
        <ListItem>
            <Icon src={calendarIcon} alt="Date published" title="Date published" />
            <time pubdate="pubdate">{date}</time>
        </ListItem>
        <ListItem>
            <Icon src={categoryIcon} alt="Category" title="Category" />
            {categoryTitle}
        </ListItem>
        <ListItem>
            <Icon src={clockIcon} alt="Time to read" title="Time to read" />
            {timeToRead} min. read
        </ListItem>
    </List>
);
