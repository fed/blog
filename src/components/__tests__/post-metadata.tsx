import React from 'react';
import renderer from 'react-test-renderer';

import { PostMetadata } from '../post-metadata';

describe('PostMetadata', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<PostMetadata date="April 21, 2021" categoryTitle="General" timeToRead={2} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
