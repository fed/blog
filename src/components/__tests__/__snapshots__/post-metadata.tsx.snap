// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostMetadata renders correctly 1`] = `
<ul
  className="sc-bdvvaa egwCKV"
>
  <li
    className="sc-gsDJrp kIyLNX"
  >
    <img
      alt="Date published"
      className="sc-dkPtyc cirgGZ"
      src="test-file-stub"
      title="Date published"
    />
    <time
      pubdate="pubdate"
    >
      April 21, 2021
    </time>
  </li>
  <li
    className="sc-gsDJrp kIyLNX"
  >
    <img
      alt="Category"
      className="sc-dkPtyc cirgGZ"
      src="test-file-stub"
      title="Category"
    />
    General
  </li>
  <li
    className="sc-gsDJrp kIyLNX"
  >
    <img
      alt="Time to read"
      className="sc-dkPtyc cirgGZ"
      src="test-file-stub"
      title="Time to read"
    />
    2
     min. read
  </li>
</ul>
`;
